ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/StreamedQueryResource.java

```diff
@@ -224,12 +229,12 @@ private Response handleStatement(
           "Statement type `%s' not supported for this resource",
           statement.getClass().getName()));
     } catch (final TopicAuthorizationException e) {
-      return Errors.accessDeniedFromKafka(e);
+      return errorHandler.accessDeniedFromKafka(e);
     } catch (final KsqlStatementException e) {
       return Errors.badStatement(e.getRawMessage(), e.getSqlStatement());
     } catch (final KsqlException e) {
       return ErrorResponseUtil.generateResponse(
-          e, Errors.badRequest(e));
+          e, Errors.badRequest(e), errorHandler);
```

Comment:
Can we move `generateResponse` into the `Errors` class? Not sure why a separate class is needed.